server/game/gameSystems/MoveCardSystem.ts

```diff
@@ -41,10 +42,16 @@ export class MoveCardSystem<TContext extends AbilityContext = AbilityContext> ex
         if (EnumHelpers.isArena(event.card.location) && !EnumHelpers.isArena(event.destination)) {
             this.leavesPlayEventHandler(event, additionalProperties);
         } else {
-            // TODO: remove this completely if determinmed we don't need card snapshots
+            // TODO: remove this completely if determined we don't need card snapshots
             // event.cardStateWhenMoved = card.createSnapshot();
             const card = event.card as Card;
 
+            // If the card is a resource and it is ready, try to ready another resource instead
+            // and exhaust this one. This should be the desired behavior for most cases.
+            if (card.location === Location.Resource && (card as CardWithExhaustProperty).ready) {
```

Comment:
Since there are a lot of systems that move cards around, might be good to do this at a lower level. Maybe add it to `Player.moveCard()`?